docs/engineering-defaults.md

```diff
@@ -1,7 +1,151 @@
 # Engineering Defaults
```

Comment:
I would like to have a intro for Engineering defaults and why we recommend these and why it is important. What outcomes are these intended to accomplish @adrielp @acramsay @ericchapman80